evalml/pipelines/pipeline_base.py

```diff
@@ -487,3 +487,21 @@ def __eq__(self, other):
             if getattr(self, attribute) != getattr(other, attribute):
                 return False
         return True
+
+    def __str__(self):
+        return self.name
+
+    def __repr__(self):
+        rpr = f"{(type(self).__name__)}(parameters={{"
+        for component, parameters in self.parameters.items():
+            rpr = rpr + f"'{component}':{{"
+            for parameter, value in parameters.items():
+                if type(value) == str:
+                    rpr = rpr + f"'{parameter}': '{value}', "
+                elif value == float('inf') or value == float('-inf'):
+                    rpr = rpr + f"'{parameter}': float('{value}'), "
+                else:
+                    rpr = rpr + f"'{parameter}': {value}, "
+            rpr = rpr + "}, "
+        rpr = rpr + '})'
+        return rpr
```

Comment:
Same comment as for component repr, let's flatten this out, and great thinking trying nan/inf types!